packages/@react-spectrum/color/src/ColorPicker.tsx

```diff
@@ -0,0 +1,122 @@
+
+export interface SpectrumColorPickerProps extends ValueBase<string | Color, Color>, AriaLabelingProps {
+  /** A visual label for the color picker. */
+  label?: ReactNode,
+  /** The contents of the color picker popover, e.g. `<ColorEditor />`. */
+  children?: ReactNode,
+  /**
+   * The size of the color swatch.
+   * @default "M"
+   */
+  size?: 'XS' | 'S' | 'M' | 'L',
+  /**
+   * The corner rounding of the color swatch.
+   * @default "default"
+   */
+  rounding?: 'default' | 'none' | 'full'
+}
+
+function ColorPicker(props: SpectrumColorPickerProps, ref: FocusableRef<HTMLButtonElement>) {
+  let swatchRef = useRef(null);
+  let domRef = useFocusableRef(ref);
+  let labelId = useId();
+  return (
+    <AriaColorPicker {...props}>
+      <DialogTrigger type="popover" mobileType="tray" targetRef={unwrapDOMRef(swatchRef)}>
+        <Button 
+          ref={domRef}
+          className={style({
+            backgroundColor: 'transparent',
+            borderStyle: 'none',
+            padding: 0,
+            display: 'flex',
+            alignItems: 'center',
+            gap: 'text-to-control',
+            outlineStyle: 'none',
+            fontFamily: 'sans',
+            color: 'body',
+            fontSize: {
+              size: {
+                XS: 'xs',
+                S: 'sm',
+                M: 'base',
+                L: 'lg'
+              }
+            }
+          })({size: props.size || 'M'})}>
+          {({isFocusVisible}) => (
+            <>
+              <div
+                className={style({
+                  outlineStyle: {
+                    default: 'none',
+                    isFocusVisible: 'solid'
+                  },
+                  outlineColor: 'focus-ring',
+                  outlineWidth: 2,
+                  outlineOffset: 2,
+                  borderRadius: 'default'
+                })({isFocusVisible})}>
+                <ColorSwatch
+                  ref={swatchRef}
+                  size={props.size}
+                  rounding={props.rounding}
+                  aria-label={props['aria-label']}
+                  aria-labelledby={props['aria-labelledby']}
+                  aria-describedby={props['aria-describedby']}
+                  aria-details={props['aria-details']} />
+              </div>
+              {props.label && 
+                <span id={labelId}>{props.label}</span>
+              }
+            </>
+          )}
+        </Button>
+        <Dialog 
+          aria-labelledby={props.label ? labelId : props['aria-labelledby']}
+          aria-label={props['aria-label']}
+          UNSAFE_style={{
+            width: 'fit-content',
+            minWidth: 0,
+            margin: '0 auto' // Center within tray.
+          }}>
+          <Content 
+            UNSAFE_style={{
+              position: 'relative',
+              margin: 'calc(var(--spectrum-dialog-padding) * -1)',
+              padding: 'var(--spectrum-global-dimension-size-200)'
```

Comment:
This is a hack to reduce the padding of the popover. Ideally we'd add a density option to popover but we need to work out the API for it.